integration/scheduler.md

```diff
@@ -0,0 +1,137 @@
+       },
+       {
+           "instrument_name": "ab12",
+           "configuration": 3457
+       },
+   ],
+}
+```
+
+See the [API documentation](https://observatorycontrolsystem.github.io/assets/html/observation-portal.html#operation/createObservation) for more information.
+
+## Advanced Topics
+Above and beyond the basic role of a scheduler, we’d like to touch on some advanced topics that an observatory should consider when implementing their own scheduler.
+
+### Telescope Telemetry for Real-Time Availability
+Any scheduler implementation should be able to understand the real-time status of observatory resources. An observatory’s Telescope Control System (TCS) can provide real-time updates about its schedulability status in order for the scheduler to understand whether to schedule observations on a given resource.
+
+If real-time status is desired, a TCS should be able to regularly post telemetry data to a centralized storage space where it can be accessed by the scheduler. Las Cumbres Observatory (LCO) utilizes Elasticsearch for this purpose - and LCO’s scheduler regularly pulls data from an Elasticsearch index that the various TCS instances update.
+
+The following telemetry data specification is sufficient to allow the scheduler to know when a resource is available/unavailable for scheduling:
+
+**Datum Name**|**Value String**|**Timestamp**|**Site**|**Enclosure**|**Telescope**|
+----------------|----------------|-------------|----------|------------|-----------|
+Available for scheduling|True/False|ISO 8601 Date|Site ID|Enclosure ID|Telescope ID|
+
+
+### Checking In-Progress Observations When Scheduling
+Depending on an observatory’s TCS implementation, it may be important for the scheduler to understand when observations are in-progress so that when it produces a new schedule during an observing night, it does not schedule new observations over in-progress observations. If the TCS software receives a new schedule with an observation overlapping an in-progress observation, it may choose to abort the observation - which isn’t always ideal.
+
+One way to avoid this issue is to take into account the time range(s) of currently-executing observation(s). 
+
+To get the running observations from the observation portal:
+
+Retrieve all PENDING, IN_PROGRESS, and COMPLETED observations between now and an end time for a given site/observatory/telescope. Usually the end time can be set to: 
+
+```
+end time = now + length of time for normal scheduling run
+```
+
+Set an initial cutoff time equal to the end time of the window being considered
+
+For each observation:
+
+If: 
+1. The end time of the window is between the start and end time of the observation,
+2. The observation's end time is greater than the cutoff time,
+3. The observation is in PENDING or IN_PROGRESS state,
+
+Then:
+1. Set the cutoff time to the observation end time.
+
+Finally, the running observation will be that whose start time is less than the cutoff end time.
```

Comment:
This end section is confusing to me. Are you just trying to explain how to determine what a running observation is?

Reply:
Yeah, I basically wanted to put into words [this code](https://github.com/observatorycontrolsystem/adaptive_scheduler/blob/75ba3787e8cf95f5e26d056f0cfd4e466f2dcd0f/adaptive_scheduler/observations.py#L335) which is a little tricky to understand at first-glance

But maybe I should just link to it?

Reply:
Yeah linking it is fine, I feel like all the text is overcomplicating it a bit. I would just say something like "To determine if an observation is currently running, check if it's start time is before the current time, and it's end time is greater than the projected end time of your scheduling run, and it is in a non-terminal state (PENDING or IN_PROGRESS). You should then use that observation's end time as the earliest start time for the next scheduled observation on that resource."